### ISSUE TITLE
[nav] Make the nav tablet/mobile friendly, add styles for desktop links

### ISSUE THREAD

**Issue by data-sync-user** [mozilla/fxa#4874]
Collapse the menu into a hamburger menu. On this button, `aria-expanded` should be set to `false`, `aria-label` should read "Site navigation menu", and `aria-haspopup` should be set to `true`.

The `aria-label` on the "X" to close the menu should read as "Close menu".

When changing something based on screen size/resize, reference the `fxa-payments-server` for the media query hook and how `PlanDetails` changes.

Also make sure all desktop focus/hover states for links are present.

See https://mozilla.invisionapp.com/share/7FUIT2QWQRE#/screens/394358870

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXA-1583)
┆Issue Number: FXA-1583
